src/Raven.Server/Documents/Sharding/Handlers/Processors/ShardedAdminForbiddenUnusedIdsHandlerProcessorForGetUnusedIds.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using NuGet.Packaging;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Server.Documents.Commands.Revisions;
+using Raven.Server.Documents.Handlers.Processors;
+using Raven.Server.Documents.Handlers.Processors.Revisions;
+using Raven.Server.Documents.Sharding.Executors;
+using Raven.Server.Documents.Sharding.Handlers.Processors.Revisions;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.Json;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors;
+
+internal class ShardedAdminForbiddenUnusedIdsHandlerProcessorForGetUnusedIds : AbstractAdminForbiddenUnusedIdsHandlerProcessorForGetUnusedIds<
+    ShardedDatabaseRequestHandler, TransactionOperationContext>
+{
+    public ShardedAdminForbiddenUnusedIdsHandlerProcessorForGetUnusedIds([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+    {
+    }
+
+    protected override async Task ValidateUnusedIdsOnAllNodesAsync(HashSet<string> unusedIds, Dictionary<string, string> forbiddenIds, string databaseName,
+        CancellationToken token)
+    {
+        var op = new ShardedGetForbiddenUnusedIdsOperation(RequestHandler,
+            new ShardedGetForbiddenUnusedIdsOperation.Parameters()
+            {
+                ValidateContent = false, // We want to validate the ids content only once - only on the orchestrator, not on every shard
+                DatabaseIds = unusedIds
+            });
+        var results = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op, token);
+
+        forbiddenIds.AddRange(results);
+    }
+
+    internal readonly struct ShardedGetForbiddenUnusedIdsOperation : IShardedOperation<Dictionary<string, string>, Dictionary<string, string>>
+    {
+        private readonly ShardedDatabaseRequestHandler _handler;
+        private readonly Parameters _parameters;
+
+        public ShardedGetForbiddenUnusedIdsOperation(ShardedDatabaseRequestHandler handler, Parameters parameters)
+        {
+            _handler = handler;
+            _parameters = parameters;
+        }
+
+        public HttpRequest HttpRequest => _handler.HttpContext.Request;
+
+        public Dictionary<string, string> Combine(Dictionary<int, ShardExecutionResult<Dictionary<string, string>>> results)
```

Comment:
I don't think you need to combine here anything... each shard will validate its own topology and throw if something is wrong.
if all succeed no exception will be thrown so you will know that everything is valid

Reply:
Removed